### PR TITLE
Add docs on architecture

### PR DESCRIPTION
**Pull request by tiagonapoli**
First part of docs on toolbelt's architecture and tips when developing and debugging toolbelt

### CODE REVIEW

File: docs/maintaining.md [hunk 1]
```diff
@@ -2,15 +2,29 @@
 
 This document is for people working on VTEX as Toolbelt maintainers.
 
+**Table of Contents**
+
+- [Forcing a new version](#forcing-a-new-version)
+- [Deprecating a version](#deprecating-a-version)
+- [Changing the release message](#changing-the-release-message)
+- [Monitoring and debugging user errors](#monitoring-and-debugging-user-errors)
+
```
Comment (tiagonapoli) [inline docs/maintaining.md:11]: Almost all changes on this document is formatting, except for this table of contents and the last section, which is new: Monitoring and debugging user errors

File: docs/architecture.md [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+- [Init Hook and CLIPreTasks](#init-hook-and-clipretasks)
+- [Telemetry](#telemetry)
+- [Appendix A: The fire and forget child_process pattern](#appendix-a-the-fire-and-forget-child_process-pattern)
+
+## Entrypoint
+
+The CLI entrypoint script is placed on `bin/run`. This script is written in javascript and is
+responsible for starting up OCLIF and requiring
+[`v8-compile-cache`](https://www.npmjs.com/package/v8-compile-cache) (ideally `v8-compile-cache`
+should be `require`d before any other require or import happens), which is responsible for speeding
```
Comment (VerasThiago) [inline docs/architecture.md:15]: `require`d => `required`

File: docs/architecture.md [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+  locally until it's decided to report to the `toolbelt-telemetry` app.
+- `logs/`: This directory is used to persist logs.
+- `session/`: This directory is used to persist data regarding the user's session (account,
+  workspace, token).
+
+All of these paths are defined as constants on the `PathConstants` class.
+
+## Toolbelt config server
+
+The (`vtex.toolbelt-config-server`)[https://github.com/vtex/toolbelt-config-server] app was created
```
Comment (VerasThiago) [inline docs/architecture.md:149]: You swapped the ref syntax, is [](), not ()[]

File: docs/architecture.md [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+account, workspace `master` and to make interactions with it easier a CLI was created, the
+[`toolbelt-config-cli`](https://github.com/vtex/toolbelt-config-cli).
+
+The updated list of configurations `toolbelt-config-server` provides is available
+[here](https://github.com/vtex/toolbelt-config-server#configs).
+
+Also, `toolbelt-config-server` can be used to serve toolbelt messages and logs that need to be
+dynamic (they are frequently modified and we want all toolbelt users to see the most updated version
+of them). The messages that can be added follow a templating scheme
+((toolbelt-message-renderer)[https://github.com/vtex/toolbelt-message-renderer]) that allow us to
```
Comment (VerasThiago) [inline docs/architecture.md:161]: Same here

File: docs/architecture.md [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+toolbelt requests should properly setup this header).
+
+Check the [`toolbelt-config-server` repository](https://github.com/vtex/toolbelt-config-server) and
+[`toolbelt-config-cli` repository](https://github.com/vtex/toolbelt-config-cli) for more useful
+information.
+
+## Init Hook and CLIPreTasks
+
+OCLIF provides the feature of [lifecycle hooks](https://oclif.io/docs/hooks#lifecycle-events) and
+one of the hooks it exposes is the `init` hook. In toolbelt we use this feature very useful feature
```
Comment (VerasThiago) [inline docs/architecture.md:178]: ```In toolbelt we use this feature very useful feature``` seems incorrect

**Review by VerasThiago — APPROVED** [review]

**Pull request merged**